src/OnlineRetailPortal.Web/Validations/GetProductsRequestValidator.cs

```diff
@@ -0,0 +1,36 @@
+﻿using OnlineRetailPortal.Contracts;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using FluentValidation;
+using FluentValidation.Results;
+using OnlineRetailPortal.Core;
+
+namespace OnlineRetailPortal.Web
+{
+    public class GetProductsRequestValidator : AbstractValidator<GetProductsServiceRequest>
+    {
+        public GetProductsRequestValidator()
+        {
+            RuleFor(x => x.PagingInfo.PageNumber)
+            .Cascade(CascadeMode.StopOnFirstFailure)
+            .NotNull()
```

Comment:
pagenumber and pagesize were int in my opinion also as per contract , these fields are optional as per contract . Why we need validation on these fields then ?

Reply:
done